### DISCUSSION TITLE
Can't seem to figure out Automation with Loop Config

### DISCUSSION THREAD

**Discussion by mmalessa80** [Budibase/budibase#6713] [Help]
My setup is as follows: I have tables  AnnualPermitData, PermitList, and AnnualPermitDataPermits_Has_RelatedPermits. In my form to create a new row for AnnualPermitData I have a relationship picker for PermitList, however when I save the new row, I receive an Error 'no default value for column id'. The AnnualPermitData_Has_PermitList table has columns of AnnualPermitData_fk, PermitList_fk, and id (primary key). Since I have been unable to figure what the issue is with this setup, I decided to try to get the same accomplished using an 'Action' automation which is setup like this on the form side:

![image](https://user-images.githubusercontent.com/106758851/179074740-0aa84a53-b65b-4ad7-aec1-9c9ba5187f4f.png)

Permits is set as an array on the automation and ThisRowId is set as text. {{ State.PermitMeterPermits }} is an array in the format ['tk34tr67','er2345ut'] which are some primary key values in the PermitList table, and {{ Form.Fields.id }} would be the primary key for the new row in the AnnualPermitData table 57c396f6 in this particular example.

Then my automation from the action is setup like this:

![image](https://user-images.githubusercontent.com/106758851/179075690-2b11bc56-83fb-4ef3-acdf-278a766490b5.png)

I then deployed my application, went to the new row form for AnnualPermitData, filled in the info and clicked save. I get the toast notification that a new row was created and I get another toast notification that my automation was triggered. I go to the AnnualPermitData table and see that a new row was in fact created, however the relationship rows in the AnnualPermitDataPermits_Has_RelatedPermits did NOT get created. When I go to my 'apps' screen and click 'manage' under this application and view the 'Automation History', there is nothing there.

I was hoping that somebody could assist me with figuring out what I am doing incorrectly under either the relationship picker process in the first paragraph, or what am I missing with the automation looping? I apologize for any ranting, but I have become somewhat frustrated.

**Comment by melohagan** [comment]
Hi @mmalessa80 

I see that you have `Use values` selected in the create row step. Could you try changing that to `Use bindings` and publishing that please?

**Reply by melohagan** [reply]
Also it looks like there could be a whitespace in the *AnnualPermiDataPermits_fk* field. Between `fields` and `.ThisRowId`

**Reply by mmalessa80** [reply]
Changed it to the following:

![image](https://user-images.githubusercontent.com/106758851/179580107-dbc06fe7-0f44-4312-ac87-f5fad7c391b9.png)

Yes, there was in fact white space, thank you for catching that. Unfortunately it appears that the loop still does not execute. Ultimately I would still like to find out what the inputs should be for a loop to properly function as I will need one for another process. But, in part this question is at least somewhat related to my other issue https://github.com/Budibase/budibase/discussions/6651.

**Comment by melohagan** [comment]
Hey @mmalessa80 

I'm thinking it could be a good idea to take a step back, and focus on your issue with saving the many-to-many relationship without the use of automations.

I was successful with the following:

- Many Students to Many Classes (with a *through* table of StudentClass)

1. Create the default screens for Students.
2. Select the `students/new/row` screen.
3. Add a button to set state named `Classes`. Return an array of ids for the classes to save against. e.g. `return [3, 4]`
    This matches with geography and history:
    ![Screenshot 2022-07-19 at 10 35 00](https://user-images.githubusercontent.com/101575380/179718738-00d60a9d-68f7-4b73-9784-2640b4720ef7.png)
4. The save button will have a save row action as follows:
    ![Screenshot 2022-07-19 at 10 38 12](https://user-images.githubusercontent.com/101575380/179719422-372faed6-a340-4c2a-b2c2-36ec96ba2da9.png)
    Which will have the binding: `return $("State.Classes")`

#### Result
![Screenshot 2022-07-19 at 10 39 55](https://user-images.githubusercontent.com/101575380/179719753-343c264c-595d-4d8e-a894-d4e9f300c179.png)

![Screenshot 2022-07-19 at 10 40 12](https://user-images.githubusercontent.com/101575380/179719805-d6dc99c7-22c4-4b0c-93a8-39c5a8e080b1.png)

![Screenshot 2022-07-19 at 10 40 57](https://user-images.githubusercontent.com/101575380/179719966-9818ea7f-cab6-4b6b-b1cc-1d1ff0b6c649.png)

**Automatically inserted**
![Screenshot 2022-07-19 at 10 42 05](https://user-images.githubusercontent.com/101575380/179720201-dc81bd58-1f48-4233-967d-85de2a56b3f0.png)
